openedx/core/djangoapps/discussions/admin.py

```diff
@@ -0,0 +1,12 @@
+from django.contrib import admin
+
+from .models import DiscussionProviderConfig, LearningContextDiscussionConfig
+
+
+class DiscussionProviderConfigAdminModel(admin.ModelAdmin):
```

Comment:
I've noticed that if I set "Provider config" back to `None` in the Learning Context Discussion Config admin, then I get the following error on the discussions page:

'NoneType' object has no attribute 'course_tab_view'

Wonder if there's something easy we can do to make this a bit safer?  Either by redirecting away from the page if discussions aren't configured, or keeping users from setting the provider config back to `None` in the admin?

Reply:
Having the provider set to None is the equivalent of a config being disabled, so I have pushed a change that should mark it as disabled in that case. I'm testing it presently.

Reply:
Ah, OK I see. The discussion tab disappears when you remove the provider, but if you navigate to it directly, it will throw this error.

Reply:
I've pushed a fix for this by raising 404 if the discussion tab is disabled. 
For the config fix, I can make that in the other PR.